src/client/java/net/neoforged/neoforge/client/resources/NeoSplashManager.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.resources;
+
+import static java.time.temporal.ChronoField.DAY_OF_MONTH;
+import static java.time.temporal.ChronoField.MONTH_OF_YEAR;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.mojang.logging.LogUtils;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.JsonOps;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import java.io.IOException;
+import java.io.Reader;
+import java.time.Month;
+import java.time.MonthDay;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import net.minecraft.client.User;
+import net.minecraft.client.gui.components.SplashRenderer;
+import net.minecraft.client.resources.SplashManager;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.Resource;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.Nullable;
+
+public class NeoSplashManager extends SplashManager {
+    private static final String SPLASHES_LOCATION = "texts/splashes.json";
+    protected final Map<MonthDay, List<String>> specialSplashes = Maps.newHashMap();
+
+    public NeoSplashManager(User user) {
+        super(user);
+    }
+
+    @Override
+    protected List<String> prepare(ResourceManager resourceManager, ProfilerFiller profilerFiller) {
+        specialSplashes.clear();
+
+        List<String> list = Lists.newArrayList();
+
+        list.addAll(super.prepare(resourceManager, profilerFiller));
+        list.add("%s IS YOU");
+
+        specialSplashes.computeIfAbsent(MonthDay.of(Month.DECEMBER, 24), k -> new ArrayList<>()).add("Merry X-mas!");
```

Comment:
Hard-coding these in this way seems bad and a good way to accidentally forget to add a new one if vanilla does -- especially given how small this feature is in general. I think I see why you did it -- on one of those days, you want any modded ones registered to be able to be chosen randomly compared to the vanilla ones -- but patching `SplashManager` instead of extending would let you at least get a patch conflict while upgrading if this scenario (these changing and the hard-coding needing updating) happened.